src/Sarif.Multitool/Rules/RuleId.cs

```diff
@@ -16,5 +16,6 @@ public static class RuleId
         public const string UriBaseIdRequiresRelativeUri = "SARIF1014";
         public const string UriMustBeAbsolute = "SARIF1015";
         public const string ContextRegionRequiresRegion = "SARIF1016";
+        public const string InvalidLogicalLocationIndex = "SARIF1017";
```

Comment:
>InvalidLogicalLocationIndex [](start = 28, length = 27)

As always, I'm concerned this rule is overly granular, create one rule for any invalid index.

Reply:
Working on it. There are a total of 15 array-index-valued properties, so it will take a couple of hours:

- `logicalLocation.index`
- `logicalLocation.parentIndex`
- `artifactLocation.index`
- `artifact.parentIndex`
- `result.ruleIndex`
- `address.index`
- `address.parentIndex`
- `threadFlowLocation.index`
- `graphTraversal.resultGraphIndex`
- `graphTraversal.runGraphIndex`
- `webRequest.index`
- `webResponse.index`
- `resultProvenance.invocationIndex`
- `reportingDescriptorReference.index`
- `toolComponentReference.index`

---
In reply to: [303553749](https://github.com/microsoft/sarif-sdk/pull/1570#discussion_r303553749) [](ancestors = 303553749)